src/views/domain-workflows/domain-workflows-query-input/domain-workflows-query-input.tsx

```diff
@@ -20,6 +21,26 @@ export default function DomainWorkflowsQueryInput({
 
   const isQueryUnchanged = value && value === queryText;
 
+  const onSubmit = useCallback(() => {
+    if (!isQueryUnchanged) {
+      setValue(queryText || undefined);
+    }
+    refetchQuery();
+  }, [isQueryUnchanged, setValue, queryText, refetchQuery]);
+
+  useEffect(() => {
```

Comment:
Using form is much better than listing to the keyboard events. The form is better for accessibility & also the enter event would only submit the form if it was within the form. 

With the keydown approach this can still run even if some modal is opened.

Reply:
The `cmd+enter` is usually the way to go when the input supports multilines otherwise `Enter` is expected. Also `e.metaKey` is not widely supported

Reply:
Using a form forces the page to reload on submit. Moreover, we intend to support multiline when we add more features to the query.

Reply:
Made it a form, but we may need to revisit this when we add a fully featured query input